[lldb] Restrict TestVariableAnnotationsDisassembler.py to ELF x86_64 (skip on Windows/COFF)

UltimateForce21 opened the pull request:
The `TestVariableAnnotationsDisassembler.py` test assembles `d_original_example.s`, 
which contains ELF-specific directives such as:

- `.ident`
- `.section ".note.GNU-stack", "", @progbits`
- `.section .debug_line, "", @progbits`

These directives are not understood by COFF on Windows, so the test fails 
on the lldb-remote-linux-win builder even when running on x86_64.

This patch adds a decorator to gate the test,
- `@skipUnlessPlatform(["linux", "freebsd", "netbsd", "android"])` — 
  runs only on ELF platforms

Follow-up to #155942.

UltimateForce21 committed:
[lldb] Restrict TestVariableAnnotationsDisassembler.py to ELF x86_64 (skip on Windows/COFF)

The test assembles `d_original_example.s`, which contains ELF-specific
directives (e.g. `.ident`, `.note.GNU-stack`, `.debug_*` sections).
These fail on Windows/COFF even on x86_64.

Gate the test to x86_64 on ELF platforms (Linux/FreeBSD/NetBSD).

UltimateForce21 committed:
Formatting change to pass darker.

DavidSpickett commented on lldb/test/API/functionalities/disassembler-variables/TestVariableAnnotationsDisassembler.py:
```diff
@@ -105,6 +117,8 @@ def test_loop_reg_rotate(self):
 
     @no_debug_info_test
+    @skipIfWindows
+    @skipUnlessPlatform(["linux", "freebsd", "netbsd"])
```
We could just skipIfWindows but then there are non-Windows COFF platforms like AIX.

Tricky to get an ideal skip here because we don't have a skip if ELF. Certainly you can remove the skipIfWindows, it's implied by the skipUnlessPlatform.

You could add `"android"` there as well. Not sure it will actually work there but I see some other tests with:
```
@skipUnlessPlatform(["linux", "android", "freebsd", "netbsd"])
```
And someone from that project can remove it later if it does fail there.

DavidSpickett reviewed (commented):
You might be able to apply the skip to the class, instead of repeating it for each function. Can you try that?

One test for example does this:
```
# Windows does not allow quotes in file names.
@skipIf(hostoslist=["windows"])
@skipIfRemote
class TestGdbRemoteLibrariesSvr4Support(gdbremote_testcase.GdbRemoteTestCaseBase):
    FEATURE_NAME = "qXfer:libraries-svr4:read"
```
Certain decorators might not work in that position though, so give it a go.

UltimateForce21 committed:
Apply class-level skip to TestVariableAnnotationsDisassembler

This change moves the platform skip logic to the test class instead of
repeating it for each method.

DavidSpickett reviewed (approved):
LGTM!

DavidSpickett commented:
Btw, you have some leftover debug `print()` in the test cases. No rush but please do remove them at some point.

DavidSpickett commented:
> LLVM Buildbot has detected a new failure on builder cross-project-tests-sie-ubuntu-dwarf5

And this failure you can safely ignore :)

UltimateForce21 commented:
> Btw, you have some leftover debug `print()` in the test cases. No rush but please do remove them at some point.

Sounds good!


> > LLVM Buildbot has detected a new failure on builder cross-project-tests-sie-ubuntu-dwarf5
> 
> And this failure you can safely ignore :)

So, we are okay?
